### PR TITLE
Remove Queue TODOS, and add actual queue support. Add reminders

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/3803
https://github.com/dotnet/arcade-services/issues/3814

Also does some general improvements

### CODE REVIEW

File: src/ProductConstructionService/ProductConstructionService.Api/Api/v2018_07_16/Controllers/SubscriptionsController.cs [hunk 1]
```diff
@@ -127,64 +136,107 @@ protected async Task<IActionResult> TriggerSubscriptionCore(Guid id, int buildId
...
 
-    /// <summary>
-    ///   Trigger daily update
-    /// </summary>
-    [HttpPost("triggerDaily")]
+        /// <summary>
+        ///   Trigger daily update
+        /// </summary>
+        [HttpPost("triggerDaily")]
     [SwaggerApiResponse(HttpStatusCode.Accepted, Description = "Trigger all subscriptions normally updated daily.")]
```
Comment (premun) [inline src/ProductConstructionService/ProductConstructionService.Api/Api/v2018_07_16/Controllers/SubscriptionsController.cs:192]:
```suggestion
    /// <summary>
    ///   Trigger daily update
    /// </summary>
    [HttpPost("triggerDaily")]
    [SwaggerApiResponse(HttpStatusCode.Accepted, Description = "Trigger all subscriptions normally updated daily.")]
```

File: src/ProductConstructionService/ProductConstructionService.Api/Api/v2020_02_20/Controllers/BuildsController.cs [hunk 2]
```diff
@@ -318,7 +323,11 @@ await _context.BuildDependencies.AddRangeAsync(
 
         // Compute the dependency incoherencies of the build.
         // Since this might be an expensive operation we do it asynchronously.
-        // TODO (https://github.com/dotnet/arcade-services/issues/3814): Queue.Post<BuildCoherencyInfoWorkItem>(JToken.FromObject(buildModel.Id));
+        await _workItemProducerFactory.Create<
+            BuildCoherencyInfoWorkItem>().ProduceWorkItemAsync(new()
+        {
+            BuildId = buildModel.Id
+        });
```
Comment (premun) [inline src/ProductConstructionService/ProductConstructionService.Api/Api/v2020_02_20/Controllers/BuildsController.cs:330]: The indentation here is broken up pretty bad 😅

File: src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemProducer.cs [hunk 3]
```diff
@@ -18,4 +18,14 @@ public async Task<SendReceipt> ProduceWorkItemAsync(T payload)
         var client = _queueServiceClient.GetQueueClient(_queueName);
         return await client.SendMessageAsync(JsonSerializer.Serialize<WorkItem>(payload));
     }
+
+    /// <summary>
+    /// Puts a WorkItem into the queue, which becomes visible after the specified delay.
+    /// </summary>
+    /// <returns></returns>
+    public async Task<SendReceipt> ProduceDelayedWorkItemAsync(T payload, TimeSpan delay)
```
[the remark this answers is not in the document]
Comment (dkurepa) [reply src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemProducer.cs]: fair point, apparently that's how `SendMessageAsync` works anyway

File: src/ProductConstructionService/ProductConstructionService.Api/Api/v2020_02_20/Controllers/BuildsController.cs (hunk 2 above)
Comment (dkurepa) [reply src/ProductConstructionService/ProductConstructionService.Api/Api/v2020_02_20/Controllers/BuildsController.cs]: ops ahah, fixed now

File: src/ProductConstructionService/ProductConstructionService.Api/Api/v2018_07_16/Controllers/RepositoryController.cs [hunk 4]
```diff
@@ -225,48 +222,17 @@ public async Task<IActionResult> RetryActionAsync([Required] string repository,
                 new ApiError("That action was successful, it cannot be retried."));
         }
 
-        // TODO (https://github.com/dotnet/arcade-services/issues/3814): Queue.Post<PullRequestActionWorkItem>(PullRequestActionWorkItem.GetArguments(update));
+        // TODO https://github.com/dotnet/arcade-services/issues/3854 figure out if we even need this method
```
[the remark this answers is not in the document]
Comment (dkurepa) [reply src/ProductConstructionService/ProductConstructionService.Api/Api/v2018_07_16/Controllers/RepositoryController.cs:225]: I'm not seeing any calls for this one, or for the Subscriptions one

[the remark this answers is not in the document]
Comment (premun) [reply src/ProductConstructionService/ProductConstructionService.Api/Api/v2018_07_16/Controllers/RepositoryController.cs:225]: Should we kill this endpoint then? I am up for it

**Comment by premun** [comment]
@dkurepa let's merge your PR before mine

File: src/ProductConstructionService/ProductConstructionService.Api/Api/v2018_07_16/Controllers/RepositoryController.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (dkurepa) [reply src/ProductConstructionService/ProductConstructionService.Api/Api/v2018_07_16/Controllers/RepositoryController.cs]: I deleted their implementation, not sure if I should just commented them out, or found some attribute to disable them

**Review by premun — APPROVED** [review]

**Review by premun — DISMISSED** [review]

**Review by premun — APPROVED** [review]

**Pull request merged**